apps/projects/app/App.js

```diff
@@ -1,120 +1,49 @@
-const AUTH_URI = 'https://tps.autark.xyz/authenticate'
-const REDIRECT_URI = 'https://tps.autark.xyz/tps-auth'
-
-export const githubPopup = (popup = null) => {
-  // Checks to save some memory if the popup exists as a window object
-  if (popup === null || popup.closed) {
-    popup = window.open(
-      // TODO: Improve readability here: encode = (params: Object) => (JSON.stringify(params).replace(':', '=').trim())
-      // encode url params
-      `${GITHUB_URI}?client_id=${CLIENT_ID}&scope=public_repo&redirect_uri=${REDIRECT_URI}`,
-      // `${REDIRECT_URI}/?code=232r3423`, // <= use this to avoid spamming github for testing purposes
-      'githubAuth',
-      // TODO: Improve readability here: encode = (fields: Object) => (JSON.stringify(fields).replace(':', '=').trim())
-      `scrollbars=no,toolbar=no,location=no,titlebar=no,directories=no,status=no,menubar=no, ${getPopupDimensions()}`
-    )
-  } else popup.focus()
-  return popup
-}
-
-const getPopupDimensions = () => {
-  const { width, height } = getPopupSize()
-  const { top, left } = getPopupOffset({ width, height })
-  return `width=${width},height=${height},top=${top},left=${left}`
-}
-
-const getPopupSize = () => {
-  return { width: 650, height: 850 }
-}
+} from './store/eventTypes'
 
-const getPopupOffset = ({ width, height }) => {
-  const wLeft = window.screenLeft ? window.screenLeft : window.screenX
-  const wTop = window.screenTop ? window.screenTop : window.screenY
+import { initApolloClient } from './utils/apollo-client'
+import { getToken, getURLParam, githubPopup, STATUS } from './utils/github'
+import { CURRENT_USER } from './utils/gql-queries'
+import { ipfsAdd, computeIpfsString } from './utils/ipfs-helpers'
+import { toHex } from './utils/web3-utils'
 
-  const left = wLeft + window.innerWidth / 2 - width / 2
-  const top = wTop + window.innerHeight / 2 - height / 2
-
-  return { top, left }
-}
-
-const getURLParam = param => {
-  const searchParam = new URLSearchParams(window.location.search)
-  return searchParam.get(param)
-}
-
-const initApolloClient = token =>
-  new ApolloClient({
-    uri: 'https://api.github.com/graphql',
-    request: operation => {
-      if (token) {
-        operation.setContext({
-          headers: {
-            accept: 'application/vnd.github.starfire-preview+json', // needed to create issues
-            authorization: `bearer ${token}`,
-          },
-        })
-      }
-    },
-  })
-
-/**
- * Sends an http request to the AUTH_URI with the auth code obtained from the oauth flow
- * @param {string} code
- * @returns {string} The authentication token obtained from the auth server
- */
-const getToken = async code => {
-  const response = await fetch(`${AUTH_URI}/${code}`)
-  const json = await response.json()
-
-  return json.token
-}
+const ASSETS_URL = './aragon-ui'
 
 class App extends React.PureComponent {
   static propTypes = {
-    app: PropTypes.object.isRequired,
+    // is not required, since it comes async
```

Comment:
Could you expand on this?

Reply:
app prop was renamed to api prop, previously app prop was a required prop (with isRequired) now, it cannot be required, because first it comes as null, then asynchronously gets a value.
Because at some initial point its value is undefined then it cannot be required anymore.